### PR TITLE
kernel/binary_manager: Change a type of kernel version to float

### PR DESCRIPTION
**Pull request by jeongarmy**
Binary manager manages kernel version as float type.

### CODE REVIEW

File: os/kernel/binary_manager/binary_manager.h [hunk 1]
```diff
@@ -165,7 +165,7 @@ struct binmgr_kinfo_s {
 	uint8_t inuse_idx;
 	uint32_t part_count;
 	part_info_t part_info[KERNEL_BIN_COUNT];
-	char version[KERNEL_VER_MAX];
+	float version;
```
Comment (sunghan-chang) [inline os/kernel/binary_manager/binary_manager.h:168]: In binary_header_s, it looks double. Is it ok with float?

File: os/include/tinyara/binary_manager.h [hunk 2]
```diff
@@ -131,7 +128,7 @@ typedef struct binary_header_s binary_header_t;
 struct binary_update_info_s {
 	int available_size;
 	char name[BIN_NAME_MAX];
-	uint32_t version;
+	double version;
```
Comment (sunghan-chang) [inline os/include/tinyara/binary_manager.h:131]: This is not kernel version you mentioned in title. Could you rewrite the title or could you split the commit?

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 3]
```diff
@@ -73,7 +73,7 @@ static void binary_update_download_binary(binary_update_info_t *binary_info, int
 	binary_header_t header_data;
 	char filepath[CONFIG_PATH_MAX];
 
-	snprintf(filepath, CONFIG_PATH_MAX, "%s/%s_%d", BINARY_DIR_PATH, APP_NAME, binary_info->version);
+	snprintf(filepath, CONFIG_PATH_MAX, "%s/%s_%d", BINARY_DIR_PATH, APP_NAME, (int)binary_info->version);
```
Comment (sunghan-chang) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:76]: Why version has type casting as int? not uint32_t or not double or not float?

**Review by sunghan-chang — APPROVED** [review]

File: os/include/tinyara/binary_manager.h (hunk 2 above)
Comment (thapav) [reply os/include/tinyara/binary_manager.h:132]: I agree, I think we can have two commits based on kernel version in binary/kernel table

File: os/kernel/binary_manager/binary_manager.h [hunk 4]
```diff
@@ -149,7 +149,7 @@ struct binmgr_uinfo_s {
 	load_attr_t load_attr;
 	uint8_t load_priority[FILES_PER_BIN];
 	uint32_t bin_ver[FILES_PER_BIN];
-	char kernel_ver[KERNEL_VER_MAX];
+	float kernel_ver;
```
Comment (thapav) [inline os/kernel/binary_manager/binary_manager.h:152]: The names are a little confusing 
Currently we are using >
**kernel_ver** for "User binary data type in binary table"
**version** for "Kernel binary data type in kernel table"
I feel we can have the naming something like kversion_ktable & kversion_btable
What is your opinion ?

File: os/tools/mkbinheader.py [hunk 5]
```diff
@@ -304,7 +309,6 @@ def get_static_ram_size(bin_type):
         addr_fp.close()
         os.remove('addr_file')
 
-        addr_data = jump_addr.split(' ')
```
Comment (amandeepc) [inline os/tools/mkbinheader.py:307]: what is the logic behind this change?

File: os/tools/mkbinheader.py [hunk 6]
```diff
@@ -258,6 +258,11 @@ def get_static_ram_size(bin_type):
         binary_ram_size = int(static_ram_size) + int(dynamic_ram_size)
         binary_ram_size = roundup_power_two(binary_ram_size)
 
+    if float(kernel_ver) <= 0 :
```
Comment (jeongchanKim) [inline os/tools/mkbinheader.py:261]: How about comparing with `0.0`?

File: os/tools/mkbinheader.py [hunk 7]
```diff
@@ -165,7 +165,7 @@ def get_static_ram_size(bin_type):
 SIZE_OF_BINVER = 4
 SIZE_OF_BINRAMSIZE = 4
 SIZE_OF_MAINSTACKSIZE = 4
-SIZE_OF_KERNELVER = 8
+SIZE_OF_KERNELVER = 4
```
Comment (jeongchanKim) [inline os/tools/mkbinheader.py:168]: Please update the description also in the above.

File: loadable_apps/loadable_sample/wifiapp/binary_update.c [hunk 8]
```diff
@@ -412,7 +412,7 @@ static void binary_update_same_version_test(void)
 	binary_update_getinfo(APP_NAME, &bin_info);
 
 	/* Try to create binary file with old version */
-	ret = binary_manager_open_new_entry(APP_NAME, bin_info.version);
+	ret = binary_manager_open_new_entry(APP_NAME, (int)bin_info.version);
```
Comment (jeongchanKim) [inline loadable_apps/loadable_sample/wifiapp/binary_update.c:415]: There is no int type version. We have uint32_t or float. How about changing the `binary_manager_open_new_entry`'s param type one of uint32_t or float?

File: os/kernel/binary_manager/binary_manager.h (hunk 4 above)
Comment (jeongarmy) [reply os/kernel/binary_manager/binary_manager.h:152]: kernel and user table is already separated and this is structure for user binary.
User binary table also should keep kernel version for checking compatibility of kernel version.

File: os/tools/mkbinheader.py [hunk 9]
```diff
@@ -118,7 +118,7 @@ def get_static_ram_size(bin_type):
 # -------------------------------------------------------------------------------------
 # ------------------------------ +
 # | Kernel Version |Jump address |
-# |  (8bytes)      |  (4bytes)   |
+# |  (4bytes)      |  (4bytes)   |
```
Comment (jeongchanKim) [inline os/tools/mkbinheader.py:121]: Please update the total size also.

File: os/kernel/binary_manager/binary_manager_load.c [hunk 10]
```diff
@@ -632,7 +632,7 @@ int binary_manager_read_header(char *path, binary_header_t *header_data, bool cr
 			goto errout_with_fd;
 		}
 	}
-	bmvdbg("Binary header : %d %d %d %d %s %d %d %s %d\n", header_data->header_size, header_data->bin_type, header_data->bin_size, header_data->loading_priority, header_data->bin_name, header_data->bin_ver, header_data->bin_ramsize, header_data->kernel_ver, header_data->jump_addr);
+	bmvdbg("Binary header : %d %d %d %d %s %d %d %.1f %d\n", header_data->header_size, header_data->bin_type, header_data->bin_size, header_data->loading_priority, header_data->bin_name, header_data->bin_ver, header_data->bin_ramsize, header_data->kernel_ver, header_data->jump_addr);
```
Comment (jeongchanKim) [inline os/kernel/binary_manager/binary_manager_load.c:635]: Please check the version type.

**Review by jeongchanKim — APPROVED** [review]

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**